### PR TITLE
Migrate curl_formadd from form API to mime API (deprecated in Ubuntu Noble)

### PR DESCRIPTION
**Pull request by j-rivero**
# 🦟 Bug fix

## Summary
The new curl 8.x series deprecate part of the CURL API that is being used in fuel-tools. I was totally new to the CURL APIs but there was a good documentation on how to migrate the old form API to the new MIME API in https://curl.se/libcurl/c/libcurl-tutorial.html (search by Converting from deprecated form api to mime api) so I went ahead and try to get the thing working.

I don't see any new test failure but probably we need someone with more knowledge of fuel to review that things are actually working.

I ran the ASAN analizer but found that we already have problems on the ASAN build (although Jenkins is not reporting them probably because https://github.com/gazebosim/gz-cmake/issues/240). 

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Comment by j-rivero** [comment]
> I ran the ASAN analizer but found that we already have problems on the ASAN build (although Jenkins is not reporting them probably because [gazebosim/gz-cmake#240](https://github.com/gazebosim/gz-cmake/issues/240)).
> ##

Fixed the memory problems in https://github.com/gazebosim/gz-fuel-tools/pull/416

File: src/RestClient.cc [hunk 1]
```diff
@@ -314,8 +307,7 @@ RestResponse Rest::Request(HttpMethod _method,
   }
   else if (_method == HttpMethod::POST_FORM)
   {
-    formpost = BuildFormPost(_form);
-    curl_easy_setopt(curl, CURLOPT_HTTPPOST, formpost);
```
Comment (nkoenig) [inline src/RestClient.cc:318]: Should there be the following line here?
```
curl_easy_setopt(curl, CURLOPT_MIMEPOST, multipart);
```

File: src/RestClient.cc [hunk 2]
```diff
@@ -314,8 +307,7 @@ RestResponse Rest::Request(HttpMethod _method,
...
+    AddFormPost(multipart, _form);
```
Comment (nkoenig) [inline src/RestClient.cc:310]: Should there be the following line here?
```
curl_easy_setopt(curl, CURLOPT_MIMEPOST, multipart);
```

Comment (j-rivero) [reply src/RestClient.cc:310]: good catch! reviewing the deprecation instruction you are right.

**Pull request merged**